app/controllers/my/events_controller.rb

```diff
@@ -15,8 +15,10 @@ def register
     volunteer_event = @event.volunteer_events.find_or_initialize_by(volunteer: current_volunteer)
     if volunteer_event.new_record?
       volunteer_event.register!
+      NewEventMailer.register_volunteer(@event, @volunteer).deliver_later
```

Comment:
think you meant `current_volunteer` here instead of `@volunteer` -- It's not initialized. So you'll probably get an `Uninitialized...`